src/main/java/org/solid/testharness/http/HttpUtils.java

```diff
@@ -83,23 +88,53 @@ public static boolean isFileProtocol(final String protocol) {
 
     public static void logRequest(final Logger logger, final HttpRequest request) {
         if (logger.isDebugEnabled()) {
-            logger.debug("REQUEST {} {}", request.method(), request.uri());
-            request.headers().map().forEach((k, v) -> logger.debug("REQ HEADER {}: {}", k, v));
+            logger.debug("REQUEST  {} {}", request.method(), request.uri());
+            logHeaders(logger, request.headers().map());
         }
     }
 
     public static <T> void logResponse(final Logger logger, final HttpResponse<T> response) {
         if (logger.isDebugEnabled()) {
             logger.debug("RESPONSE {} {}", response.request().method(), response.uri());
             logger.debug("STATUS   {}", response.statusCode());
-            response.headers().map().forEach((k, v) -> logger.debug("HEADER   {}: {}", k, v));
+            logHeaders(logger, response.headers().map());
             final T body = response.body();
             if (body != null) {
-                logger.debug("BODY     {}", body);
+                logger.debug("BODY     {}", maskBody((String)body));
             }
         }
     }
 
+    private static void logHeaders(final Logger logger, final Map<String, List<String>> headerMap) {
+        headerMap.forEach((key, values)
+                -> values.forEach(value -> logger.debug("HEADER   {}: {}", key, maskHeader(key, value)))
```

Comment:
Generally I would agree but there are 2 related factors in this:
1. The HTTP interactions are quite important for API testing so they feel more significant than trace but that wouldn't stop people still turning trace on if they needed it
2. The bigger problem is that the Karate steps use DEBUG and I can't change that. I matched that level for the setup and auth flow logging but the reality is that people are more likely to only turn on those logs as the most relevant to a test and leave mine out.

So I could change setup and auth flow logging to TRACE since they are a different source to the Karate logs but might it be more confusing to have similar types of logs split across both DEBUG and TRACE?

I am inclined to leave things as they are in this case but what do you think?

Reply:
Alright, it sounds like weighty enough reasons to keep them that way.